components/atom/spinner/src/index.js

```diff
@@ -0,0 +1,115 @@
+import React, {Component} from 'react'
+import ReactDOM from 'react-dom'
+import PropTypes from 'prop-types'
+import cx from 'classnames'
+import SUILoader from './SUILoader'
+
+const TYPES = {
+  FULL: 'full',
+  SECTION: 'section'
+}
+
+const DELAY = 500 // ms
+const BASE_CLASS = 'sui-AtomSpinner'
+
+class AtomSpinner extends Component {
+  state = {
+    delayed: this.props.delayed
+  }
+
+  get _parentNodeClassList () {
+    if (this._parentNodeClassListCache) return this._parentNodeClassListCache
+
+    this._parentNodeClassListCache = ReactDOM.findDOMNode(this).parentNode.classList
+    return this._parentNodeClassListCache
+  }
+
+  get _parentClassName () {
+    const {type} = this.props
+
+    return cx(
+      type === TYPES.SECTION ? BASE_CLASS : `${BASE_CLASS}--fullPage`
+    )
+  }
+
+  componentDidMount () {
+    if (!this.state.delayed) {
+      this.props.show && this._addParentClass()
+      return
+    }
+
+    this.timer = setTimeout(() => {
+      this.setState({delayed: false}, this._addParentClass)
+    }, DELAY)
+  }
+
+  componentWillUnmount () {
+    clearTimeout(this.timer)
+    this._removeParentClass()
+  }
+
+  _removeParentClass () {
+    this._parentNodeClassList.remove(this._parentClassName)
+  }
+
+  _addParentClass () {
+    this._parentNodeClassList.add(this._parentClassName)
+  }
+
+  componentWillReceiveProps (nextProps) {
+    if (this.props.show === nextProps.show) return
+    if (!this.props.show) return this._addParentClass()
+    if (this.props.show) return this._removeParentClass()
+  }
+
+  shouldRenderLoader () {
+    const {show} = this.props
+    const {delayed} = this.state
+
+    return !delayed && show
```

Comment:
I'm sorry but I just one last thing.
Why do we even need a show prop ? Does it make sense to set show to true or false instead of simple not rendering the spinner.


I mean, instead of 
````
<AtomSpinner show={this.state.show} type={AtomSpinnerTypes.FULL} />
````


this:
````
this.state.show && <AtomSpinner type={AtomSpinnerTypes.FULL} />
````

I'll approve the PR because I guess there's a reason but I wanna know it :-)

Reply:
No, there is no reason, very well seen